### ISSUE TITLE
Getting a lot of Log abandoned Log reacquired

### ISSUE THREAD

**Issue by unisqu** [varnishcache/varnish-cache#2470]
Log abandoned
Log reacquired
No idea what it means or how to debug. Can anyone advise where should I look for more information?

**Comment by nigoroll** [comment]
see also #2460 and #2430

**Comment by unisqu** [comment]
If i dont update this part, is it a serious bug? what kind of problems do i expect to get? i'm reloading varnish quite frequently.

**Comment by simonvik** [comment]
I'm seeing the same problem in 5.2, reloading varnish pretty often.

**Comment by unisqu** [comment]
before anything, is it a critical bug to operation?

**Comment by nigoroll** [comment]
TL;DR: If you see panics, yes, Otherwise, no

In principle, `Log abandoned & Log reacquired` should only happen when varnish is panicking. So if you see the `MGT.child_panic` statistic > 0 and/or `varnishadm panic.show` succeeds, you are experiencing panics. Then the root cause is different to the issue discussed in the two tickets I referenced and we should look after the panics themselves based on the output of `varnishadm panic.show` (that would be a new ticket please).
The issue we have newly introduced with 5.2 which is discussed in the referenced tickets has, with as much certainty as a human can give you, zero impact on delivery. It affects log clients and the worst I have seen so far is #2460 where a log client would hang, reporting Log overrun indefinitely. commit 17d7c4f at least avoids the hang. With this patch, I am not aware of any relevant impact, but I would suspect that this issue may lead to some log information being missed by vsl readers (varnishlog / varnishncsa etc.)

**Comment by nigoroll** [comment]
As suspected, this is related to VSC destroys. Also, when I run varnishstat with the test below, I see the Hitrate_n resets documented in #2430 

Simple to reproduce with the small `vmod_debug` extension I've just pushed.
* install `vmod_debug.so` manually
* vcl:
```
vcl 4.0;
import debug;
backend default { .host = "127.0.0.2"; }
sub vcl_recv {
	if (req.url == "/new") {
		debug.vsc_new();
		return(synth(200));
	} else if (req.url == "/destroy") {
		debug.vsc_destroy();
		return(synth(200));
	}
	return(synth(404));
}
```
* simple enough test:

`$ /tmp/bin/varnishlog -g raw -i ReqURL & for i in {1..10} ; do curl -sI localhost:8080/{new,destroy} | grep HTTP  ; sleep 1 ; done ; pkill varnishlog
`
```
[4] 17612
HTTP/1.1 200 OK
HTTP/1.1 200 OK
[3]   Done                    /tmp/bin/varnishlog -g raw -i ReqURL
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
        55 ReqURL         c /new
        56 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
     32828 ReqURL         c /new
     32829 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
        58 ReqURL         c /new
        59 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
     32831 ReqURL         c /new
     32832 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
        61 ReqURL         c /new
HTTP/1.1 200 OK
        62 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
     32834 ReqURL         c /new
     32835 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
        64 ReqURL         c /new
        65 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
     32837 ReqURL         c /new
     32838 ReqURL         c /destroy
Log abandoned
Log reacquired
HTTP/1.1 200 OK
HTTP/1.1 200 OK
        67 ReqURL         c /new
        68 ReqURL         c /destroy
Log abandoned
Log reacquired
[4]+  Done                    /tmp/bin/varnishlog -g raw -i ReqURL

```

**Comment by hermunn** [comment]
Backport review: This is the new VSM implementation, so backporting is not not necessary.
